CMakeLists.txt

```diff
@@ -144,52 +144,25 @@ string(CONCAT OPEN3D_VERSION
 project(Open3D VERSION ${OPEN3D_VERSION} LANGUAGES C CXX)
 message(STATUS "Open3D ${PROJECT_VERSION}")
 
-# Detect process or architecture.
-# Ref: https://stackoverflow.com/a/12024211/1255535
-set(detect_arch_code "
-#if defined(__x86_64) || defined(__x86_64__) || defined(__amd64) || \
-        defined(_M_X64)
-#define PROCESSOR_ARCH \"x86_64\"
-#elif defined(__ARMv8_A__) || defined(__ARM64_ARCH_8__) || \
-        defined(__aarch64__) || defined(__ARMv8__) || defined(_M_ARM64)
-#define PROCESSOR_ARCH \"armv8\"
-#elif defined(__ARM_ARCH_7__) || defined(__ARM_ARCH_7A__) ||    \
-        defined(__ARM_ARCH_7R__) || defined(__ARM_ARCH_7M__) || \
-        defined(__ARM_ARCH_7EM__)
-#define PROCESSOR_ARCH \"armv7\"
-#else
-#define PROCESSOR_ARCH \"unknown\"
-#endif
-
-#include <iostream>
-int main() {
-    std::cout << PROCESSOR_ARCH;
-    return 0;
-}
-")
-file(WRITE "${CMAKE_BINARY_DIR}/detect_arch.cpp" "${detect_arch_code}")
-try_run(
-    dummy_run_result                       # <runResultVar>
-    dummy_compile_result                   # <compileResultVar>
-    "${CMAKE_BINARY_DIR}/bin"              # <bindir>
-    "${CMAKE_BINARY_DIR}/detect_arch.cpp"  # <srcfile>
-    RUN_OUTPUT_VARIABLE PROCESSOR_ARCH
-)
-if("${PROCESSOR_ARCH}" STREQUAL "unknown")
-    message(FATAL_ERROR "Unsupported processor architecture.")
-else()
-    message(STATUS "Detected processor architecture: ${PROCESSOR_ARCH}.")
+if(UNIX AND NOT APPLE)
+    execute_process(COMMAND uname -p
```

Comment:
`CMAKE_HOST_SYSTEM_PROCESSOR` contains output of `uname -m`.
https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html

Reply:
done.